File: set7/4/Main.java
import java.util.*;
import java.lang.*;
import java.io.*;

public class Main {
    public static void main(String[] args) {
        FastReader rd = new FastReader();

        int N = rd.nextInt();
        int []arr = new int[N];
        for(int i = 0; i < N; ++i) arr[i] = rd.nextInt();
        int L = 0, R = N - 1;
        int ans = 0;
        while(L < R) {
            ans = Math.max(ans, (R - L - 1) * Math.min(arr[L], arr[R]));
            if(arr[L] <= arr[R]) L ++;
            else R --;
        }
        System.out.println(ans);
    }

    static class FastReader {
        BufferedReader br;
        StringTokenizer st;

        public FastReader() {
            br = new BufferedReader(new InputStreamReader(System.in));
        }

        String next() {
            while(st == null || !st.hasMoreElements()) {
                try {
                    st = new StringTokenizer(br.readLine());
                }
                catch (IOException e) {
                    e.printStackTrace();
                }
            }
            return st.nextToken();
        }

        int nextInt() { return Integer.parseInt(next()); }
        long nextLong() { return Long.parseLong(next()); }
        double nextDouble() { return Double.parseDouble(next()); }
        String nextLine() {
            String str = "";
            try {
                str = br.readLine();
            }
            catch (IOException e) {
                e.printStackTrace();
            }
            return str;
        }
    }
}

File: set7/4/main.cpp
#include<iostream>
#include<vector>
#include<algorithm>

using namespace std;

int main() {
    ios::sync_with_stdio(false);
    cin.tie(0);

    int N; cin >> N;
    vector<int> V(N);
    for(int i= 0; i < N; ++i) cin >> V[i];

    int L = 0, R = N - 1;
    int ans = 0;
    while(L < R) {
        ans = max(ans, (R - L - 1) * min(V[L], V[R]));
        if(V[L] <= V[R]) ++L;
        else --R;
    }
    cout << ans;

    return 0;
}